addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1047 @@
+        self.env['calendar.alarm_manager'].get_next_mail()
+
+    def get_recurrent_date_by_event(self):
+        """Get recurrent dates based on Rule string and all event where recurrent_id is child
+        """
+
+        def todate(date):
+            val = parser.parse(''.join((re.compile('\d')).findall(date)))
+            ## Dates are localized to saved timezone if any, else current timezone.
+            if not val.tzinfo:
+                val = pytz.UTC.localize(val)
+            return val.astimezone(timezone)
+
+        timezone = pytz.timezone(self.env.context.get('tz') or 'UTC')
+        startdate = fields.Datetime.context_timestamp(self.with_context(tz=timezone), fields.Datetime.from_string(self.start))
+        rset1 = rrule.rrulestr(str(self.rrule), dtstart=startdate, forceset=True)
+        all_events = self.with_context(active_test=False).search([('recurrent_id', '=', self.id)])
+        for ev in all_events:
+            rset1._exdate.append(todate(ev.recurrent_id_date))
+        return [d.astimezone(pytz.UTC) for d in rset1]
+
+    def _get_recurrency_end_date(self):
+        if not self.recurrency:
+            return False
+        if self.end_type == 'count' and self.count and self.rrule_type and self.stop:
+            count = self.count + 1
+            delay, mult = {
+                'daily': ('days', 1),
+                'weekly': ('days', 7),
+                'monthly': ('months', 1),
+                'yearly': ('years', 1),
+            }[self.rrule_type]
+
+            deadline = fields.Datetime.from_string(self.stop)
+            return deadline + relativedelta(**{delay: count * mult})
+        return self.final_date
+
+    def _find_my_attendee(self):
+        """
+            Return the first attendee where the user connected has been invited from all the meeting_ids in parameters
+        """
+        for event in self:
+            for attendee in event.attendee_ids:
+                if self.env.user.partner_id.id == attendee.partner_id.id:
+                    return attendee
+        return False
+
+    def get_date_formats(self):
+        lang = self.env.lang
+        lang_params = {}
+        if lang:
+            res_lang = self.env['res.lang'].search([("code", "=", lang)])
+            if res_lang:
+                lang_params = {
+                    'date_format': res_lang.date_format,
+                    'time_format': res_lang.time_format}
+
+        # formats will be used for str{f,p}time() which do not support unicode in Python 2, coerce to str
+        format_date = lang_params.get("date_format", '%B-%d-%Y').encode('utf-8')
+        format_time = lang_params.get("time_format", '%I-%M %p').encode('utf-8')
+        return (format_date, format_time)
+
+    def get_display_time_tz(self, tz=False):
+        context = self.env.context.copy()
+        if tz:
+            context["tz"] = tz
+        return self.with_context(context)._get_display_time(self.start, self.stop, self.duration, self.allday)
```

Comment:
self.with_context(tz=tz or self.env.context.get('tz'))... and check if you can remove line 139 to 141